stl/inc/iterator

```diff
@@ -1037,14 +1037,17 @@ struct incrementable_traits<common_iterator<_Iter, _Se>> {
     using difference_type = iter_difference_t<_Iter>;
 };
 
-template <class>
+template <class, class>

```

Comment:
This is a primary template - it looks like this because the usage is `_Common_iterator_pointer_type<_Iter, _Se>` - see the constrained partial specialization below (right lines 1046-1048). (`int = 0` typically appears when we have something that doesn't really want to be a template, but has to for various reasons, like deferring instantiation until it's needed. Here, this struct does want to be a template.)

Perhaps we should consistently name template parameters in primary templates, even when they aren't needed in the primary itself. (This is my personal preference, and historical practice in the STL, but I've been attempting to be less nitpicky so I haven't pushed for this as there isn't a clear safety rationale, just tension between "write clear code" and "don't write unnecessary code", and names are indeed not strictly necessary here.)

Reply:
Allright, that's fine then.